contracts.ncl

```diff
@@ -275,6 +275,7 @@ from the Nix world) or a derivation defined in Nickel.
         | default
         = {},
       "%{type_field}" | force = "nickelDerivation",
+      ..
```

Comment:
Should we define `packages` here?

Reply:
That's a good question. I think not, because `packages` is something defined by `BashShell`, so we shouldn't have to know about it here. Similarely, we'll want other shells to be able to extend what we put in here (say have `RustShell` give you a `rustPackage` version to override the one used), so this thing should be arbitrarily extensible.

That raises an interesting issue as I don't know how we can ergonomically represent a closed, but extensible record contract.

Reply:
Actually, def4628 makes this Okay-ish for the moment, so we can stick with it I guess

Reply:
Agree, it looks much better :)